### ISSUE TITLE
LSTM different sequence length between training and validation, negative dimension error!

### ISSUE THREAD

**Issue by MatthewD1993** [fchollet/keras#7461]
I have a LSTM task, input data is inertial sensor and gyroscope measurements, 6D. Output is relative displacement 7D (3D pos, 4D angular).
    Now I use 70% data for training, and training sequence is 200, batch size is 1, and set the LSTM to be stateful because I want it to remember the states from the last sequence.  I write a customized loss function, owing to the fact the last 4D of output should always have norm 1 (Quaternion angle representation, not important to this question)
    After every epoch, I want to measure the loss with validation data set, that is a single very long sequence, can I do it like this?  **Validation and training sequence have different lengths.**
But I got error like this
**InvalidArgumentError:** Shape [-1,-1,6] has negative dimensions

I attached detailed **Python notebook** as followed:
https://github.com/MatthewD1993/CameraPose/blob/master/my_exp/sensor_fusioin_LSTM-Copy1.ipynb

- Another question is: 
I write a metric function return a vector 7D, but in the logs seems only the first value is returned. Any idead?

Thanks in advance!

**Comment by td2014** [comment]
@MatthewD1993 .  I'm trying to see if I understand what your training data shape is.  The typical input that I believe Keras expects for LSTM (or general recurrent networks) is something like:
(batch_size, num_timesteps, input_features).  Are you saying that you have training data shape is (1,200,6) or something else?  Just trying to get a handle on your parameters which might suggest some direction.  In any case, you might want to check that you are providing an input shape to your first layer which indicates the number of timesteps, and number of input features.  The convention in Keras in to NOT include the size of the batch dimension.  I hope this helps a bit.  Thanks.
